src/cmd/completion.go

```diff
@@ -43,13 +42,15 @@ func completionEmpty(_ *cobra.Command, _ []string, _ string) ([]string, cobra.Sh
 }
 
 func completionCommands(cmd *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
-	commandNames := []string{}
+	// init-container is hidden but documented command that is meant to be
+	// accessed by users at least via a manual page
+	commandNames := []string{"init-container"}
```

Comment:
Anyway, when it comes to shell completion, I think that we should skip both `completion` and `init-container`.  We don't expect users to directly invoke either of them.  eg., `toolbox-init-container(1)` says:
```
It is not expected to be directly invoked by humans
```

Having it show up in the completion will only encourage enterprising souls to mess with it, leading to unforeseen consequences.

I see that it [used to be](https://github.com/containers/toolbox/commit/8b84b5e4604921fad818ede5267591de7f0993e6) part of the older hand-written competion file.  So, dropping it would be a change in behaviour, but I think it's right direction.

Reply:
I second the opinion to not show `init-container` in the completion suggestions but due to the completion of the `help` command being related more towards the immediate help of the user.

We document, should keep on documenting and should propagate the way we bootstrap toolbxes. It is a vital part of the project and deserves some explanation into its inner workings.

As a result the last commit can be dropped from the PR.

Reply:
I forgot what the final conclusion for this discussion was.

As I mentioned [before](https://github.com/containers/toolbox/pull/1055#discussion_r966183914), `init-container` can be important for those who are trying to troubleshoot or poke around at their Toolbxes.  It shows up in `podman inspect` and does have some expectation of stability.  That's why we want to document it.

However, we also don't want casual users start messing with `init-container`.  That's why we shouldn't expose it in the shell completion for `toolbox init-co<TAB>`, but I suppose it's fine to expose it for `toolbox help init-co<TAB>`.  However, I don't know if it's feasible to make that distinction through Cobra.